enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+---
+title: optionally-deploy-csi-plugin
+authors:
+  - "@copejon"
+reviewers:
+  - "@pacevedom: MicroShift team-lead"
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+  - "@jakobmoellerdev Edge Enablement LVMO SME"
+  - "@suleymanakbas91,  Edge Enablement LVMO SME"
+approvers:
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+api-approvers:
+  - "None"
+creation-date: 2024-06-10
+last-updated: 2024-06-11
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-856
+see-also:
+  - "enhancements/microshift/microshift-default-csi-plugin.md"
+---
+
+# Optionally Deploy CSI Plugin
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized by
+tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See
+[kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+Out of the box, MicroShift includes the LVMS CSI driver and CSI snapshotter. The LVM Operator is not itself packaged
+with the platform, in adherence with the project's [guiding principles](./kubernetes-for-device-edge.md#goals). See the
+[MicroShift Default CSI Plugin](microshift-default-csi-plugin.md) proposal for a more in-depth explanation of the
+storage provider and reasons for its integration into MicroShift. Configuration of the driver is exposed via a config
+file at /etc/microshift/lvmd.yaml. The manifests required to deploy and configure the driver and snapshotter are baked
+into the MicroShift binary during compilation and are deployed during runtime. This means that even if a user wanted to
+disable the driver and/or snapshotter, they would be deployed again when the service restarts. Additionally, the LVMS an
+CSI containers required to run LVMS are always packaged with the product and thus will always consume a certain amount
+of storage space devices where MicroShift is deployed.
```

Comment:
This storage space equivalent to ~60kB uncompressed, in the binary it is even less than that due to binary compression. Im not sure if this is even a relevant point.

Reply:
I'm confused as to where the ~60Kb value comes from.  Note that the sentence contains a typo and should have said "LVMS and CSI container _images_", referring to the size of the images on disk.

Reply:
Ah, well, I was talking about the manifests, clearly didnt read that at full. Sorry for the confusion :) We will have only 2 images in the future, one for lvms and 1 for csi snapshotter, where the base layers will be shared with the release payload. I at expecting at most 100mi gain, but that is still relevant.